packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FLTGoogleMapHeatmapController.m

```diff
@@ -0,0 +1,186 @@
+}
+
+- (void)setGradient:(GMUGradient *)gradient {
+  _heatmapTileLayer.gradient = gradient;
+}
+
+- (void)setOpacity:(double)opacity {
+  _heatmapTileLayer.opacity = opacity;
+}
+
+- (void)setRadius:(int)radius {
+  _heatmapTileLayer.radius = radius;
+}
+
+- (void)setMinimumZoomIntensity:(int)intensity {
+  _heatmapTileLayer.minimumZoomIntensity = intensity;
+}
+
+- (void)setMaximumZoomIntensity:(int)intensity {
+  _heatmapTileLayer.maximumZoomIntensity = intensity;
+}
+
+- (void)setMap {
+  _heatmapTileLayer.map = _mapView;
+}
+
+- (void)interpretHeatmapOptions:(NSDictionary<NSString *, id> *)data {
+  id weightedData = data[kHeatmapDataKey];
+  if ([weightedData isKindOfClass:[NSArray class]]) {
+    [self setWeightedData:[FLTGoogleMapJSONConversions weightedDataFromArray:weightedData]];
+  }
+
+  id gradient = data[kHeatmapGradientKey];
+  if ([gradient isKindOfClass:[NSDictionary class]]) {
+    [self setGradient:[FLTGoogleMapJSONConversions gradientFromDictionary:gradient]];
+  }
+
+  id opacity = data[kHeatmapOpacityKey];
+  if ([opacity isKindOfClass:[NSNumber class]]) {
+    [self setOpacity:[opacity doubleValue]];
+  }
+
+  id radius = data[kHeatmapRadiusKey];
+  if ([radius isKindOfClass:[NSNumber class]]) {
+    [self setRadius:[radius intValue]];
+  }
+
+  id minimumZoomIntensity = data[kHeatmapMinimumZoomIntensityKey];
+  if ([minimumZoomIntensity isKindOfClass:[NSNumber class]]) {
+    [self setMinimumZoomIntensity:[minimumZoomIntensity intValue]];
+  }
+
+  id maximumZoomIntensity = data[kHeatmapMaximumZoomIntensityKey];
+  if ([maximumZoomIntensity isKindOfClass:[NSNumber class]]) {
+    [self setMaximumZoomIntensity:[maximumZoomIntensity intValue]];
+  }
+
+  // The map must be set each time for options to update.
+  [self setMap];
+}
+- (NSDictionary<NSString *, id> *)getHeatmapInfo {
+  NSMutableDictionary *options = [@{} mutableCopy];
+  options[kHeatmapDataKey] =
+      [FLTGoogleMapJSONConversions arrayFromWeightedData:_heatmapTileLayer.weightedData];
+  options[kHeatmapGradientKey] =
+      [FLTGoogleMapJSONConversions dictionaryFromGradient:_heatmapTileLayer.gradient];
+  options[kHeatmapOpacityKey] = @(_heatmapTileLayer.opacity);
+  options[kHeatmapRadiusKey] = @(_heatmapTileLayer.radius);
+  options[kHeatmapMinimumZoomIntensityKey] = @(_heatmapTileLayer.minimumZoomIntensity);
+  options[kHeatmapMaximumZoomIntensityKey] = @(_heatmapTileLayer.maximumZoomIntensity);
+  return options;
+}
+@end
+
+@interface FLTHeatmapsController ()
+
+// A map from heatmapId to the controller that manages it.
+@property(nonatomic, strong) NSMutableDictionary *heatmapIdToController;
+
+// The map view owned by GoogmeMapController.
+@property(nonatomic, weak) GMSMapView *mapView;
+
+@end
+
+@implementation FLTHeatmapsController
+- (instancetype)initWithMapView:(GMSMapView *)mapView {
+  self = [super init];
+  if (self) {
+    _mapView = mapView;
+    _heatmapIdToController = [@{} mutableCopy];
```

Comment:
It is better to declare the thing you want directly than to declare something else, create a copy of it, and then throw the original away; this is making the code less performant at runtime, every single time it's run, to save a few keystrokes one time.